renderdoc/driver/gl/gl_counters.cpp

```diff
@@ -148,48 +278,54 @@ vector<CounterResult> GLReplay::FetchCounters(const vector<uint32_t> &counters)
 
   MakeCurrentReplayContext(&m_ReplayCtx);
 
-  uint32_t counterID = counters[0];
-  RDCASSERT(counters.size() == 1);
-  RDCASSERT(counterID == eCounter_EventGPUDuration);
-
-  SCOPED_TIMER("Fetch Counters for %u", counterID);
-
   CounterContext ctx;
 
   for(int loop = 0; loop < 1; loop++)
   {
     ctx.eventStart = 0;
     ctx.reuseIdx = loop == 0 ? -1 : 0;
-    FillTimers(ctx, m_pDriver->GetRootDraw());
+    m_pDriver->SetFetchCounters(true);
+    FillTimers(ctx, m_pDriver->GetRootDraw(), counters);
+    m_pDriver->SetFetchCounters(false);
 
     double nanosToSecs = 1.0 / 1000000000.0;
 
     GLuint prevbind = 0;
     m_pDriver->glGetIntegerv(eGL_QUERY_BUFFER_BINDING, (GLint *)&prevbind);
     m_pDriver->glBindBuffer(eGL_QUERY_BUFFER, 0);
 
-    for(size_t i = 0; i < ctx.timers.size(); i++)
+    for(size_t i = 0; i < ctx.queries.size(); i++)
     {
-      if(ctx.timers[i].obj)
-      {
-        GLuint elapsed = 0;
-        m_pDriver->glGetQueryObjectuiv(ctx.timers[i].obj, eGL_QUERY_RESULT, &elapsed);
-
-        double duration = double(elapsed) * nanosToSecs;
-
-        ret.push_back(CounterResult(ctx.timers[i].eventID, counterID, duration));
-      }
-      else
+      for(uint32_t c = 0; c < counters.size(); c++)
       {
-        ret.push_back(CounterResult(ctx.timers[i].eventID, counterID, 0.0));
+        if(ctx.queries[i].obj[counters[c]])
+        {
+          GLuint64 data = 0;
+          m_pDriver->glGetQueryObjectui64v(ctx.queries[i].obj[counters[c]], eGL_QUERY_RESULT, &data);
+
+          if(m_pDriver->glGetError())
+            data = 3440434404;
```

Comment:
Oh right, I didn't see that at all!

Currently these counters aren't actually being printed in the UI anywhere anyway right? I think having -1 as an error value works because then it's easy to check for in code as an invalid value and return it. Although note it would need to be converted to -1.0f for floating point data types.